core/primitives-core/src/code.rs

```diff
@@ -1,4 +1,6 @@
-use near_primitives_core::hash::{hash as sha256, CryptoHash};
+use std::fmt::{Debug, Formatter};
+
+use crate::hash::{hash as sha256, CryptoHash};
 
 pub struct ContractCode {
```

Comment:
I am intentionally not making this a protocol schema and instead implementing conversion between this and CodeBytes, since we may need to change this later (eg. replace its field types with CodeHash and CodeBytes)

Reply:
I'm not sure I fully follow the reasoning here. Other members of `DistributeStateWitnessRequest` such as `ShardChunkHeader` and `ChunkStateWitness` derive `ProtocolSchema`. Isn't it a good practice to make `ContractCode` `ProtocolSchema` as well?

Reply:
I believe `ContractCode` will become protocol schema eventually. I did not want to make it yet, because I am defining new types to represent code hash and code itself, which are different from what `ContractCode` uses. I would like to wait to do the refactoring for `ContractCode` to use them, before making it a protocol schema struct.

Also `DistributeStateWitnessRequest` is an internal message from client actor to partial witness actor, so we do not need to make it protocol schema.